include/nbl/video/SPhysicalDeviceFeatures.h

```diff
@@ -0,0 +1,508 @@
+    //VkBool32           shaderBufferFloat32AtomicMinMax;
+    //VkBool32           shaderBufferFloat64AtomicMinMax;
+    //VkBool32           shaderSharedFloat16Atomics;
+    //VkBool32           shaderSharedFloat16AtomicAdd;
+    //VkBool32           shaderSharedFloat16AtomicMinMax;
+    //VkBool32           shaderSharedFloat32AtomicMinMax;
+    //VkBool32           shaderSharedFloat64AtomicMinMax;
+    //VkBool32           shaderImageFloat32AtomicMinMax;
+    //VkBool32           sparseImageFloat32AtomicMinMax;
+    
+    /* DemoteToHelperInvocationFeaturesEXT *//* VK_EXT_shader_demote_to_helper_invocation *//* MOVED TO Vulkan 1.3 Core */
+
+    /* ShaderImageAtomicInt64FeaturesEXT *//* VK_EXT_shader_image_atomic_int64 */
+    //VkBool32           shaderImageInt64Atomics;
+    //VkBool32           sparseImageInt64Atomics;
+
+    /* SubgroupSizeControlFeaturesEXT *//* VK_EXT_subgroup_size_control *//* MOVED TO Vulkan 1.3 Core */
+
+    /* TexelBufferAlignmentFeaturesEXT *//* VK_EXT_texel_buffer_alignment */
+    //VkBool32           texelBufferAlignment;
+
+    /* TextureCompressionASTCHDRFeaturesEXT *//* VK_EXT_texture_compression_astc_hdr *//* MOVED TO Vulkan 1.3 Core */
+
+    /* TransformFeedbackFeaturesEXT *//* VK_EXT_transform_feedback */
+    //VkBool32           transformFeedback;
+    //VkBool32           geometryStreams;
+
+    /* VertexAttributeDivisorFeaturesEXT *//* VK_EXT_vertex_attribute_divisor */
+    //VkBool32           vertexAttributeInstanceRateDivisor;
+    //VkBool32           vertexAttributeInstanceRateZeroDivisor;
+
+    /* VertexInputDynamicStateFeaturesEXT *//* VK_EXT_vertex_input_dynamic_state */
+    //VkBool32           vertexInputDynamicState;
+
+    /* Ycbcr2Plane444FormatsFeaturesEXT *//* VK_EXT_ycbcr_2plane_444_formats */
+    //VkBool32           ycbcr2plane444Formats;
+
+    /* YcbcrImageArraysFeaturesEXT *//* VK_EXT_ycbcr_image_arrays */
+    //VkBool32           ycbcrImageArrays;
+
+    /* ShaderIntegerFunctions2FeaturesINTEL *//* VK_INTEL_shader_integer_functions2 */
+    //VkBool32           shaderIntegerFunctions2;
+
+    /* 16BitStorageFeaturesKHR *//* VK_KHR_16bit_storage *//* MOVED TO Vulkan 1.1 Core */
+    /* 8BitStorageFeaturesKHR *//* VK_KHR_8bit_storage *//* MOVED TO Vulkan 1.2 Core */
+
+    /* AccelerationStructureFeaturesKHR *//* VK_KHR_acceleration_structure */
+    bool accelerationStructure = false;
+    bool accelerationStructureCaptureReplay = false;
+    bool accelerationStructureIndirectBuild = false;
+    bool accelerationStructureHostCommands = false;
+    bool descriptorBindingAccelerationStructureUpdateAfterBind = false;
+            
+    /* BufferDeviceAddressFeaturesKHR *//* VK_KHR_buffer_device_address *//* MOVED TO Vulkan 1.2 Core */
+
+    /* DynamicRenderingFeaturesKHR *//* VK_KHR_dynamic_rendering *//* MOVED TO Vulkan 1.3 Core */
+
+    /* [!!NV Version below, struct doesn't exist in vk headers] VK_KHR_fragment_shader_barycentric */
```

Comment:
`VkBool32 fragmentShaderBarycentric`

Reply:
true if the NV extension is present on GL

(the AMD one has really poor semantics which don't exactly map 1:1)

Reply:
you can always enable and just report as a limit